processor/src/main/java/org/mapstruct/ap/internal/conversion/BigDecimalToPrimitiveConversion.java

```diff
@@ -50,11 +50,15 @@ public String getToExpression(ConversionContext conversionContext) {
 
     @Override
     public String getFromExpression(ConversionContext conversionContext) {
-        return "BigDecimal.valueOf( <SOURCE> )";
+        return bigDecimal( conversionContext ) + ".valueOf( <SOURCE> )";
     }
 
     @Override
     protected Set<Type> getFromConversionImportTypes(ConversionContext conversionContext) {
         return asSet( conversionContext.getTypeFactory().getType( BigDecimal.class ) );
     }
+
+    private String bigDecimal(ConversionContext conversionContext) {
```

Comment:
This function is the same on 5 different places. Maybe can move it to the parent? Or have a util?

Reply:
Yes I thought the same but wasn't sure where to put.. I found a util package, maybe create a util class with static methods?

Reply:
Yes you can put them there. `ConversionUtils` maybe?

Reply:
Ok great, I will move all those helper methods that are spread over the `SimpleConversion` classes to such a new class.